### PR TITLE
Fixed parameters and function calls

### PR DESCRIPTION
**Pull request by adibalaram**
I've tried to fix the function calls in TPI.py. I do have a few questions though so if someone could tell me what to do, I can make subsequent changes to the file.

1. Lines 32 and 38 define r_11 and r_path, both of which had function calls to firm.get_r. What values of L and K are supposed to be passed to these two functions?

2. Lines 68 and 75 had K_1 and BQ_1. I have changed these to K_11 and BQ_11. Is that right?

3. What are b_1, BQ_init and new_BQ?

It would be really helpful if someone could add comments to the code to explain what is being done in this file.

### CODE REVIEW

**Comment by miltonash** [comment]
@adibalaram The function calls to firm.get_r to get market clearing conditions used for r_11 and r_path. You are correct regarding the typos you caught in Lines 68 and 75. Also, I should have commented in the code that the p loop solves saving and populates the upper-right corner of the b_matrix (twisted donut) and the t loop solves saving for the rest of full lifetimes.

**Comment by saharnazb** [comment]
I think that:

1- It seems you already did it and I agree with you on that. 
2- Yes, that seems to be a good and right value to start the path
3- From what I understand, b_1 is the initial saving, the [0] value of saving vector in the twist donut. I think in the next step the other values of bmat are appended to this (replaced in `bmat= np.zeros`! (Or maybe I am wrong) I am working on the rest

Milton and I are working on debugging that because it misses so many things.

File: scripts/TPI.py [hunk 1]
```diff
@@ -22,22 +22,30 @@ def solve_tp(g_n_path, omega_S_preTP, rho_s, imm_rates_path, params):
...
+        # I can't figure out how r_11 and r_path differ
+        ''' Is r_11 an initial guess? Depending on how you're defining
+        r_11, the arguments passed to the function call above and below
+        will vary
+        ''' 
+        r_path = firm.get_r(L_ss, K_11, alpha, A, delta)
+        w_path = firm.get_w(r_path, alpha, A, delta)
         bmat = np.zeros((S - 1, T + S - 1))
+        # What is b_1 supposed to be?
         bmat[:, 0] = b_1
```
Comment (saharnazb) [inline scripts/TPI.py:42]: add `nmat[: , 0] = n_1` #I want to include n path in here too

**Comment by adibalaram** [comment]
For my first point, which values of K and L are to be passed to the two get_r function calls? I will let y'all fix the third point and we've now confirmed the second point.

File: scripts/TPI.py (hunk 1 above)
Comment (adibalaram) [reply scripts/TPI.py:42]: But what's `n_1`? Do you mean `n_1 = nmat[: , 0]`?

Comment (jdebacker) [reply scripts/TPI.py:42]: Right. This n is endogenous. You’ll need to solve for it from the person in there last year of life who chooses labor supply but who will choose zero savings.

File: scripts/TPI.py [hunk 2]
```diff
@@ -22,22 +22,30 @@ def solve_tp(g_n_path, omega_S_preTP, rho_s, imm_rates_path, params):
...
 
         # Solve for households
```
Comment (saharnazb) [inline scripts/TPI.py:44]: ### Solve for partial lifetime decision-makers: (please someone check the periods in here; It's 1 AM in the morning here and I cannot really concentrate very well -- I defined b one period a head by adding plus 1 and kept n in S - p)
nmat = np.zeros((S, T + S - 1))

The all following is inside the loop. I do not know how to put the spaces in comments. One more thing I want to add is whether to create b_n_init using `.stack` or `.append`. I got confused there. Is there a difference?


for p in range(2, S):
            b_init = np.diag(bmat[S - p + 1:, :p - 1])
            n_init = np.diag(nmat[S - p:, :p - 1])
            b_n_init = np.hstack((b_init, n_init))
            b_n_args = (r_path[:p-1], w_path[:p-1], beta, sigma, BQpath_init[:p], rho_s[-p:], b_ellipse, upsilon)
            results_b_n_p = opt.root(hh.FOCs, b_n_init, args=(b_n_args))
            b_solve_p = results_b_n.x[:p - 1]
            n_solve_p = results_b_n.x[p - 1:]
            DiagMaskb = np.eye(p - 1, dtype=bool)
            DiagMaskn = np.eye(p - 1, dtype=bool) 
            bmat[S - p + 1:, 1:p] = DiagMaskb * b_solve_p + bmat[S - p + 1:, 1:p]
            nmat[S - p:, :p - 1] = DiagMaskn * n_solve_p + nmat[S - p:, :p-1]

File: scripts/TPI.py [hunk 3]
```diff
@@ -22,22 +22,30 @@ def solve_tp(g_n_path, omega_S_preTP, rho_s, imm_rates_path, params):
...
         for p in range(2, S):
```
Comment (saharnazb) [inline scripts/TPI.py]: I think we can do the same here for n when we solve for the full life decision maker

File: scripts/TPI.py [hunk 4]
```diff
@@ -47,7 +55,7 @@ def solve_tp(g_n_path, omega_S_preTP, rho_s, imm_rates_path, params):
         for t in range(1, T + 1):
```
Comment (saharnazb) [inline scripts/TPI.py:55]: Sorry I meant here for the t loop
I think we can do the same here for n when we solve for the full life decision maker

File: scripts/TPI.py (hunk 2 above)
Comment (miltonash) [reply scripts/TPI.py:44]: I think `.stack` and `.append` are both built-on `.concatenate`. So no real difference to my knowledge. `.append` might flatten an array if axis isn’t specified though. I like your use of `.hstack`.

**Pull request merged**